### PR TITLE
feat: [crescent] store liquid staking state

### PR DESCRIPTION
**Pull request by huichiaotsou**
## Description

Closes: #XXXX

Store liquid staking state for calculating the APR.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: modules/liquidstaking/handle_periodic_operations.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	"github.com/go-co-op/gocron"
+	"github.com/rs/zerolog/log"
+
+	"github.com/forbole/bdjuno/v3/modules/utils"
+)
+
+// RegisterPeriodicOperations implements modules.PeriodicOperationsModule
+func (m *Module) RegisterPeriodicOperations(scheduler *gocron.Scheduler) error {
+	log.Debug().Str("module", "distribution").Msg("setting up periodic tasks")
```
Comment (MonikaCat) [inline modules/liquidstaking/handle_periodic_operations.go]:
```suggestion
	log.Debug().Str("module", "liquidstaking").Msg("setting up periodic tasks")
```

File: modules/liquidstaking/handle_periodic_operations.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+	// Update the liquid staking state every 1 day
+	if _, err := scheduler.Every(1).Day().At("00:00").Do(func() {
+		utils.WatchMethod(m.getLatestLiquidStakingState)
+	}); err != nil {
+		return fmt.Errorf("error while scheduling distribution peridic operation: %s", err)
```
Comment (MonikaCat) [inline modules/liquidstaking/handle_periodic_operations.go]:
```suggestion
		return fmt.Errorf("error while scheduling liquidstaking peridic operation: %s", err)
```

File: modules/liquidstaking/utils_liquid_states.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"fmt"
+
+	"github.com/forbole/bdjuno/v3/types"
+	"github.com/rs/zerolog/log"
+)
+
+// updateLiquidStakingState fetch the liquid staking state and store it into database
+func (m *Module) updateLiquidStakingState(height int64) error {
+	log.Debug().Str("module", "distribution").Int64("height", height).Msg("getting liquid staking state")
```
Comment (MonikaCat) [inline modules/liquidstaking/utils_liquid_states.go]:
```suggestion
	log.Debug().Str("module", "liquidstaking").Int64("height", height).Msg("getting liquid staking state")
```

File: modules/liquidstaking/utils_liquid_states.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+	state, err := m.source.GetLiquidStakingStates(height)
+	if err != nil {
+		return fmt.Errorf("error while getting liquid staking state: %s", err)
+	}
+
+	// Store the pool into the database
```
Comment (MonikaCat) [inline modules/liquidstaking/utils_liquid_states.go]:
```suggestion
	// Store the liquid staking state in the database
```

File: types/liquidstaking.go [hunk 5]
```diff
@@ -15,3 +17,17 @@ func NewLiquidStakingParams(params liquidstakingtypes.Params, height int64) *Liq
 		Height: height,
 	}
 }
+
+// LiquidStakingState represents the x/liquidstaking parameters
```
Comment (MonikaCat) [inline types/liquidstaking.go]:
```suggestion
// LiquidStakingState represents the x/liquidstaking state
```

**Review by MonikaCat — APPROVED** [review]
tested, LGTM

**Review by rmontagnin — APPROVED** [review]
utACK

**Pull request merged**